contracts/L2/gateway/L2Migrator.sol

```diff
@@ -111,6 +114,8 @@ contract L2Migrator is L2ArbitrumMessenger, IMigrator {
                 _params.delegate
             );
 
+            IDelegatorPool(poolAddr).initialize(bondingManagerAddr);
```

Comment:
Let's update the L2Migrator unit test to check for this call.

Reply:
added in addc66e

However, had to tweak it a little bit because of the weird behaviour. 
```ts
expect(DelegatorMock.initialize).to.be.called
```
is failing when using a mock contract. 

I suspect it has something do with the clone function. So deployed a real DelegatorPool Implementation and created a real contract to test if the bondingManager and Migrator addresses are set correctly after initialize function is called.

Tried different combinations of real and fake contracts but couldn't get it to work any other way. 
Would like to know your thoughts.

Reply:
> I suspect it has something do with the clone function.

Yeah I'm guessing that the reason is that a smock fake is only able to keep track of function calls that happen *after* the fake is created at an address so in this case since the `initialize()` call happens in `finalizeMigrateDelegator()` *before* the smock fake is created the fake is not aware of the `initialize()` call.

The approach that you took with a real DelegatorPool implementation looks good to me. Let's just remove the commented out smock fake code for the DelegatorPool.